src/mapping/BarycentricBaseMapping.cpp

```diff
@@ -0,0 +1,101 @@
+
+#include <Eigen/Core>
+#include <algorithm>
+#include <deque>
+#include <memory>
+#include <ostream>
+#include <unordered_set>
+#include <utility>
+
+#include "logging/LogMacros.hpp"
+#include "mapping/BarycentricBaseMapping.hpp"
+#include "mapping/Polation.hpp"
+#include "math/differences.hpp"
+#include "mesh/Data.hpp"
+#include "mesh/Mesh.hpp"
+#include "mesh/SharedPointer.hpp"
+#include "mesh/Vertex.hpp"
+#include "query/Index.hpp"
+#include "utils/Event.hpp"
+#include "utils/Statistics.hpp"
+#include "utils/assertion.hpp"
+
+namespace precice {
+extern bool syncMode;
+
+namespace mapping {
+
+BarycentricBaseMapping::BarycentricBaseMapping(Constraint constraint, int dimensions)
+    : Mapping(constraint, dimensions)
+{
+}
+
+bool BarycentricBaseMapping::hasComputedMapping() const
+{
+  return _hasComputedMapping;
+}
+
+void BarycentricBaseMapping::clear()
+{
+  PRECICE_TRACE();
+  _interpolations.clear();
+  _hasComputedMapping = false;
+}
+
+void BarycentricBaseMapping::map(
+    int inputDataID,
+    int outputDataID)
+{
+  PRECICE_TRACE(inputDataID, outputDataID);
+
+  precice::utils::Event e("map.bm.mapData.From" + input()->getName() + "To" + output()->getName(), precice::syncMode);
+
+  mesh::PtrData          inData    = input()->data(inputDataID);
+  mesh::PtrData          outData   = output()->data(outputDataID);
+  const Eigen::VectorXd &inValues  = inData->values();
+  Eigen::VectorXd &      outValues = outData->values();
+  //assign(outValues) = 0.0;
```

Comment:
I know that this is probably a copy relic, but I would grab the opportunity to clean it up a bit.
```suggestion

```